store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -38,10 +39,38 @@ public class OwaspHtmlSanitizer implements Callable<String> {
     private String vHost;
 
     public OwaspHtmlSanitizer(String html, boolean neuterImages, String vHost) {
-        this.html = html;
+        // fix to check extra tags & remove
+        this.html = checkUnbalancedTags(html);
         this.neuterImages = neuterImages;
         this.vHost = vHost;
     }
+    
+    /** 
+     * A method to check unbalanced tags which may have started, not ended or removed
+     * 
+     * @return the Formatted Html after removing unbalanced tags
+     */
+    private String checkUnbalancedTags(String html) {
+        ArrayList<String> tags = new ArrayList<String>();
+        // in tags array we can add multiple tags, so for each tag write start & end with a & separator like(<!--&-->)
+        tags.add("<!--&-->");
+        String formattedHtml = "";
+        for (String str:tags) {
+            formattedHtml = "";
```

Comment:
Already assigned in line 57 as  String formattedHtml = "";.
59 remaines redundant.

Reply:
removed 2nd occurance